src/checks/#cc4a#check_in_iteration.clas.abap

```diff
@@ -0,0 +1,347 @@
+      if analyzer->is_bracket( current_token ) eq analyzer->bracket_type-opening.
+        token_index = analyzer->calculate_bracket_end( statement = statement_to_negate bracket_position = token_index ).
+        continue.
+      endif.
+      if current_token-lexeme = `AND` or current_token-lexeme = `OR`.
+        if amount_of_concatenation > 1.
+          use_not = abap_true.
+          exit.
+        endif.
+        amount_of_concatenation += 1.
+      endif.
+      if analyzer->token_is_comparison_operator( token = current_token ).
+        insert token_index into table positions_comparison_operators.
+      endif.
+      if token_index <> lines( statement_to_negate-tokens ).
+        token_index += 1.
+      endif.
+    endloop.
+    negate_struct-use_not = use_not.
+    negate_struct-comperasion_operators = positions_comparison_operators.
+  endmethod.
+
+  method negate_statement.
+
+    data(logic_negate) = determine_negation_kind( analyzer = analyzer statement_to_negate = statement ).
+
+    data(statement_to_negate) = statement.
+
+    if logic_negate-use_not = abap_false.
+      loop at logic_negate-comperasion_operators into data(operator_position).
+        data(token_to_negate) = statement_to_negate-tokens[ operator_position ].
+        if token_to_negate-lexeme = `IS` and statement_to_negate-tokens[ operator_position + 1 ]-lexeme = `NOT`.
+          delete statement_to_negate-tokens index operator_position + 1.
+        elseif token_to_negate-lexeme = `NOT` and statement_to_negate-tokens[ operator_position + 1 ]-lexeme = `IN`.
+          delete statement_to_negate-tokens index operator_position.
+        else.
+          token_to_negate-lexeme = analyzer->negate_comparison_operator( comparison_operator = token_to_negate-lexeme ).
+          statement_to_negate-tokens[ operator_position ] = token_to_negate.
+        endif.
+      endloop.
+    else.
+      insert value #( lexeme = `NOT (` ) into statement_to_negate-tokens index 2.
+      insert value #( lexeme = ` )` ) into statement_to_negate-tokens index lines( statement_to_negate-tokens ) + 1.
+    endif.
+
+    new_statement = statement_to_negate.
+  endmethod.
+
+  method create_quickfix_code_for_if.
+    data(new_statement) = statement.
+    data(analyzer) = /cc4a/abap_analyzer=>create(  ).
+
+    new_statement-keyword = `IF`.
+    new_statement-tokens[ 1 ]-lexeme = `IF`.
+
+    data(flat_new_statement) = analyzer->flatten_tokens( negate_statement( analyzer = analyzer statement = new_statement )-tokens ) && `.`.
+    modified_statement = analyzer->break_into_lines( flat_new_statement ).
+
+    insert `CONTINUE .` into modified_statement index 2.
+    insert `ENDIF .` into modified_statement index 3.
+  endmethod.
+
+  method create_quickfix_code_for_where.
+    data(new_statement) = statement.
+    data(analyzer) = /cc4a/abap_analyzer=>create(  ).
+    data(bool_expression) = ``.
+    data(cut_of_var) = ``.
```

Comment:
Should this be something like `component`? Otherwise it's still at least a typo, i.e. should be `cut_off_var`.